src/message_extensions.rs

```diff
@@ -48,14 +52,28 @@ impl Hash for PbftMessageInfo {
     }
 }
 
+impl Hash for PbftBlock {
+    fn hash<H: Hasher>(&self, state: &mut H) {
```

Comment:
Does this clean up a clippy lint?

Reply:
Yes - clippy was upset that I was using the protobuf-derived `PartialEq` but providing my own version of `Hash` (`derive_hash_xor_eq`). I made certain that the property `k1 == k2 ==>  hash(k1) == hash(k2)` holds - see comment at the top of the file.

Reply:
This commit should be broken up more for each change that was needed as it is hard to understand with all these changes together